testing/src/KokkosFFT_CountErrors.hpp

```diff
@@ -0,0 +1,514 @@
+          Layout>::outer_iteration_pattern;
+  static const Kokkos::Iterate inner_iteration_pattern =
+      Kokkos::Impl::layout_iterate_type_selector<
+          Layout>::inner_iteration_pattern;
+  using iterate_type =
+      Kokkos::Rank<2, outer_iteration_pattern, inner_iteration_pattern>;
+  using policy_type = Kokkos::MDRangePolicy<ExecutionSpace, iterate_type,
+                                            Kokkos::IndexType<iType>>;
+
+  ///\brief Constructs a FindErrors object and performs the error detection.
+  /// This constructor initializes the output error views and launches a
+  /// parallel kernel to scan through the input view elements. For each element,
+  /// if the difference between the two views exceeds the specified tolerance,
+  /// the corresponding error values and their index are recorded.
+  ///
+  ///\param a [in] The first Kokkos view containing data.
+  ///\param b [in] The second Kokkos view containing data to compare against.
+  ///\param nb_errors [in] The maximum number of errors expected.
+  ///\param rtol [in] Relative tolerance for the comparison (default 1.e-5).
+  ///\param atol [in] Absolute tolerance for the comparison (default 1.e-8).
+  ///\param space [in] The execution space used to launch the parallel kernel.
+  FindErrors(const AViewType& a, const BViewType& b, const iType nb_errors,
+             double rtol = 1.e-5, double atol = 1.e-8,
+             const ExecutionSpace space = ExecutionSpace())
+      : m_a(a),
+        m_b(b),
+        m_a_error("a_error", nb_errors),
+        m_b_error("b_error", nb_errors),
+        m_loc_error("loc_error", nb_errors, 3),
+        m_count("count"),
+        m_rtol(rtol),
+        m_atol(atol) {
+    Kokkos::parallel_for("FindErrors-2D",
+                         policy_type(space, {0, 0}, {a.extent(0), a.extent(1)}),
+                         *this);
+  }
+
+  /// \brief Executes the element-wise comparison for the given index.
+  /// This operator is invoked in parallel by Kokkos. For each index, it
+  /// compares the corresponding elements from the two views. If the absolute
+  /// difference exceeds the tolerance, it stores the error values and their
+  /// index.
+  ///
+  /// \param i0 [in] The index along the first dimension of the element in the
+  /// views.
+  /// \param i1 [in] The index along the second dimension of the element in the
+  /// views.
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const iType i0, const iType i1) const {
+    auto tmp_a     = m_a(i0, i1);
+    auto tmp_b     = m_b(i0, i1);
+    bool not_close = are_not_close(tmp_a, tmp_b, m_rtol, m_atol);
+    if (not_close) {
+      std::size_t count     = Kokkos::atomic_load(m_count.data());
+      m_a_error(count)      = tmp_a;
+      m_b_error(count)      = tmp_b;
+      m_loc_error(count, 0) = i0 + i1 * m_a.extent(0);
+      m_loc_error(count, 1) = i0;
+      m_loc_error(count, 2) = i1;
+      Kokkos::atomic_fetch_add(m_count.data(), 1);
```

Comment:
Same remark as above
```suggestion
      std::size_t count     = Kokkos::atomic_fetch_add(m_count.data(), 1);
      m_a_error(count)      = tmp_a;
      m_b_error(count)      = tmp_b;
      m_loc_error(count, 0) = i0 + i1 * m_a.extent(0);
      m_loc_error(count, 1) = i0;
      m_loc_error(count, 2) = i1;
```